sdk/couchbase/tests/collection_crud.rs

```diff
@@ -565,3 +568,408 @@ fn get_with_expiry() {
         assert_eq!("test", content);
     })
 }
+
+#[test]
+fn get_with_projections() {
+    run_test(async |cluster| {
+        let collection = cluster
+            .bucket(cluster.default_bucket())
+            .scope(cluster.default_scope())
+            .collection(cluster.default_collection());
+
+        let content = std::fs::read_to_string("tests/testdata/projection_doc.json").unwrap();
+        let person: Person = serde_json::from_str(content.as_str()).unwrap();
+
+        let projection_id = new_key();
+        collection
+            .upsert(&projection_id, &person, None)
+            .await
+            .unwrap();
+
+        #[derive(Debug)]
+        struct TCase {
+            name: &'static str,
+            project: Vec<String>,
+            expected: Person,
+        }
+
+        let cases: Vec<TCase> = vec![
```

Comment:
Unsure about this, if any of these cases fail then it'll halt the execution of the rest. Which could be annoying?